libs/plugin/src/generators/convert-outputs/generator.ts

```diff
@@ -0,0 +1,316 @@
+				throw `"${project}" project not found`;
+			}
+
+			visitNotIgnoredFiles(tree, projectConfiguration.root, (path) => {
+				trackContents(tree, contentsStore, path);
+			});
+		} catch (err) {
+			logger.error(`[ngxtension] ${err}`);
+			return;
+		}
+	} else {
+		const projects = getProjects(tree);
+		for (const project of projects.values()) {
+			visitNotIgnoredFiles(tree, project.root, (path) => {
+				trackContents(tree, contentsStore, path);
+			});
+		}
+	}
+
+	for (const { path: sourcePath } of contentsStore.collection) {
+		const sourceFile = contentsStore.project.getSourceFile(sourcePath);
+
+		const hasOutputImport = sourceFile.getImportDeclaration(
+			(importDecl) =>
+				importDecl.getModuleSpecifierValue() === '@angular/core' &&
+				importDecl
+					.getNamedImports()
+					.some((namedImport) => namedImport.getName() === 'output'),
+		);
+
+		// NOTE: only add hasOutputImport import if we don't have it and we find the first Output decorator
+		if (!hasOutputImport) {
+			sourceFile.addImportDeclaration({
+				namedImports: ['output'],
+				moduleSpecifier: '@angular/core',
+			});
+		}
+
+		const classes = sourceFile.getClasses();
+
+		for (const targetClass of classes) {
+			const applicableDecorator = targetClass.getDecorator((decoratorDecl) => {
+				return ['Component', 'Directive'].includes(decoratorDecl.getName());
+			});
+			if (!applicableDecorator) continue;
+
+			const convertedOutputs = new Set<string>();
+
+			let outputFromObservableImportAdded = false;
+
+			targetClass.forEachChild((node) => {
+				if (Node.isPropertyDeclaration(node)) {
+					const outputDecorator = node.getDecorator('Output');
+					if (outputDecorator) {
+						const {
+							name,
+							isReadonly,
+							docs,
+							scope,
+							hasOverrideKeyword,
+							initializer,
+						} = node.getStructure();
+
+						const {
+							needsOutputFromObservableImport,
+							removeOnlyDecorator,
+							outputName,
+							writerFn,
+						} = getOutputInitializer(
+							name,
+							outputDecorator,
+							initializer as string,
+						);
+
+						if (
+							needsOutputFromObservableImport &&
+							!outputFromObservableImportAdded
+						) {
+							sourceFile.addImportDeclaration({
+								namedImports: ['outputFromObservable'],
+								moduleSpecifier: '@angular/core/rxjs-interop',
+							});
+							outputFromObservableImportAdded = true;
+						}
+
+						const newProperty = targetClass.addProperty({
+							name: outputName ?? name,
+							isReadonly,
+							docs,
+							scope,
+							hasOverrideKeyword,
+							initializer: writerFn,
+						});
+
+						if (removeOnlyDecorator) {
+							outputDecorator.remove();
+
+							newProperty.setHasExclamationToken(false);
+
+							newProperty.addJsDoc(
+								'TODO(migration): you may want to convert this to a normal output',
+							);
+						} else {
+							node.replaceWithText(newProperty.print());
+							// remove old class property Output
```

Comment:
after `node.replaceWithText(newProperty.print());` 
this code is needed for something like 'this.someOutput.next() to be replaced with this.someOutput.emit()' as this was actually possible on EventEmiter

```
newProperty.findReferencesAsNodes().forEach((reference) => {
								try {
									//we are looking for something like 'this.someOutput.next() as we want to replace it with this.someOutput.emit()'
									//for each identifier we go someOutput, this.someOutput , then DotNotation , then Indentifier and we check that this identifier is 'next'
									const nextIdentifier = reference
										.getParentIfKindOrThrow(SyntaxKind.PropertyAccessExpression)
										.getNextSiblingIfKindOrThrow(SyntaxKind.DotToken)
										.getNextSiblingIfKindOrThrow(SyntaxKind.Identifier);
									if (nextIdentifier.getText() === 'next') {
										if (
											nextIdentifier
												.getParent()
												.getParent()
												.isKind(SyntaxKind.CallExpression)
										) {
											console.log(
												nextIdentifier.getParent().getParent().getText(),
											);
											console.log('\nFOUND next - replacng with emit');
											nextIdentifier.replaceWithText('emit');
										}
									}
								} catch (err) {}
							});
```


I have large project that I'm migrating so I will be adding coments like this if you are ok with it with next things that I stumbled on

now working on types as EventEmitter() is typed EventEmitter<any> but output() is OutputEmitterRef<void>, also found usages of accessing valueChanges on eventEmitters etc so will take care of those as well

Reply:
of course here simple regex could be used but I feel like this is much safer in terms of migration

Reply:
Hello @kbrilla 
Thanks for all this 🙌🙏
Would you mind send a PR that we can merge into this PRs branch? 
And then we can merge this PR into main and cut a release?